### ISSUE TITLE
Group charter for 2024

### ISSUE THREAD

**Issue by RupertBenWiser** [WebView-CG/charter#12]
This issue will be used to track high level discussions into the charter for 2024.

The chairs are proposing that we add the deliverable, a webview compatibility report. This would serve as a reference point into the differences between different WebView providers.

We are also proposing to add a decision process like other community groups have.

**Comment by RupertBenWiser** [comment]
Just some ideas for the compatibility report if we go in that direction. Trying to maintain compatibility with caniuse sounds like it could be a great idea:
https://github.com/Fyrd/caniuse/blob/main/CONTRIBUTING.md

**Comment by RupertBenWiser** [comment]
Pull request with proposed charter changes:
https://github.com/WebView-CG/charter/pull/13

We will discuss this in the next group meeting.

**Comment by tidoust** [comment]
> Just some ideas for the compatibility report if we go in that direction. Trying to maintain compatibility with caniuse sounds like it could be a great idea: https://github.com/Fyrd/caniuse/blob/main/CONTRIBUTING.md

More specifically, a useful approach could perhaps be to base this compatibility report on [web-features](https://github.com/web-platform-dx/web-features#web-features-exploring-the-set-of-interoperable-features-of-the-web-platform), a shared catalog of features of the web platform developed in the WebDX Community Group, with representatives from browser vendors, MDN, BCD, Can I Use (this includes @Elchi3, @Fyrd and me).

That would allow to maintain compatibility with Can I Use, but also with MDN (both Can I Use and MDN now use web-features to document baseline support), and hopefully with other places that need to talk about web features at a relatively coarse level, down the road. It would also probably help with bringing WebView support data to BCD directly.

**Comment by tidoust** [comment]
Expanding on today's discussion, what I think could be useful for developers is not a one-time document that describes the API differences between WebViews, but rather continuously updated support data about WebViews that get integrated and reflected in available documentation, in turn to provide incentives to maintain the data over time.

From a charter perspective, instead of a "document", the WebView CG could perhaps endeavor to gather compatiblity data for WebViews, feed them into common sources (best one that comes to mind would indeed be [BCD](https://github.com/mdn/browser-compat-data/), since it is used in MDN, Can I Use, and web-features), and develop or contribute to tools that can ease data maintenance.

This could mean adjusting #13 to:
- Replace the "WebView compatibility report" line with "WebView compatibility data and tools: make sure machine-readable data describing the API differences between different WebView & browser implementations are available to documentation platforms and other tools, by February 2025." (Not necessarily a good  fit for a section entitled "Non-Normative Reports", perhaps, oh well ;))
- Complete the first bullet in the scope (which start with "identify the issues") with somethine like "including gathering support data to document interoperable surface of WebViews". Or create a separate bullet. An update may not be needed.

On top of gathering the data, which would also be a problem to create a document, the main hurdle with that proposal is of course data maintenance... Hence the idea to create synergies between people who have support data and those who can consume it.

BCD contains fine-grained features. If it it turns out to be impractical to provide support data at that level, this is where [web-features](https://github.com/web-platform-dx/web-features) could help, as it provides a list of higher-level features that are more directly usable in discourse, and that map to BCD features (meaning a tool could easily and automatically convert the higher-level support data for use in BCD).

**Comment by RupertBenWiser** [comment]
Thanks, those suggestions are great and I absolutely agree with the sentiment of focusing on cementing process rather than having a one time report.

I figured it can't hurt to just add a second heading called non-normative data?

It would also be great to investigate if the WebView providers can help support this data programmatically to some degree. This deliverable captures that, while not necessarily committing us to it if it proves too difficult.

I've updated the charter proposal PR. If we move ahead with this updated charter, I'll create a new repo that we can start tracking these TODOs in.
